### PR TITLE
Pass optional kwargs through to assessment functions

### PR DESCRIPTION
**Pull request by woodwardmw**
## Summary
- Add `kwargs: Optional[Dict[str, Any]]` field to `AssessmentIn` model with validation (max 20 keys, valid Python identifiers, scalar values only, string cap 1000 chars)
- Accept `extra_kwargs` as an optional query parameter (JSON-encoded string) in `POST /assessment` endpoint
- Persist kwargs to DB via new `kwargs` JSONB column for audit trail (alembic migration `eca4f540a198`)
- Replace deprecated `.dict()` with `.model_dump()` in `call_assessment_runner`

**Companion PR:** sil-ai/aqua-assessments#125 threads kwargs through the runner to Modal assess() calls with reserved key filtering.

## Test plan
- [x] Run existing tests to verify no regressions (3/3 passing)
- [x] Verify validator fires correctly (scalar-only, key name, key count)
- [ ] Verify OpenAPI docs show extra_kwargs as optional query parameter
- [ ] Test with a semantic-similarity assessment passing extra kwargs
- [ ] Add dedicated test cases for kwargs validation edge cases

### CODE REVIEW

File: assessment_routes/v3/assessment_routes.py [hunk 1]
```diff
@@ -199,11 +200,14 @@ async def call_assessment_runner(assessment: AssessmentIn, return_all_results: b
 @router.post("/assessment", response_model=List[AssessmentOut])
 async def add_assessment(
     a: AssessmentIn = Depends(),
+    extra_kwargs: Optional[str] = Query(
+        None,
+        alias="extra_kwargs",
```
[the remark this answers is not in the document]
Comment (woodwardmw) [reply assessment_routes/v3/assessment_routes.py]: Agreed — the alias was redundant since the variable and query param share the same name. Removed it.
